crt0stack/src/entry.rs

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/// An auxiliary vector entry
+#[derive(Debug, PartialEq)]
+pub enum Entry<'a> {
+    /// file descriptor of program
+    ExecFd(usize), // core does not have RawFd
+
+    /// program headers for program
+    PHdr(usize),
+
+    /// size of program header entry
+    PHent(usize),
+
+    /// number of program headers
+    PHnum(usize),
+
+    /// system page size
+    PageSize(usize),
+
+    /// base address of interpreter
+    Base(usize),
+
+    /// flags
+    Flags(usize),
+
+    /// entry point of program
+    Entry(usize),
+
+    /// program is not ELF
+    NotElf(bool),
+
+    /// real uid
+    Uid(usize),
+
+    /// effective uid
+    EUid(usize),
+
+    /// real gid
+    Gid(usize),
+
+    /// effective gid
+    EGid(usize),
+
+    /// string identifying CPU for optimizations
+    Platform(&'a str),
+
+    /// arch dependent hints at CPU capabilities
+    HwCap(usize),
+
+    /// frequency at which times() increments
+    ClockTick(usize),
+
+    /// secure mode boolean
+    Secure(bool),
+
+    /// string identifying real platform, may differ from Platform.
+    BasePlatform(&'a str),
```

Comment:
What is a "real platform"?

Reply:
So, some 64bit platforms offer a 32bit mode, which then is the `Platform`. The `BasePlatform` would be the 64bit one.

Reply:
Interesting... thanks, @haraldh !